data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,256 @@
+  }
+
+  def validateEncryptionRequest(request: JobRequest): Boolean = {
+    if (request.encryption_key.nonEmpty) true else false;
+  }
+
+  def validateRequest(request: JobRequest): Boolean = {
+    if (request.request_data != null) true else false
+  }
+
+  def markRequestAsProcessing(request: JobRequest):Boolean = {
+    request.status = "PROCESSING";
+    updateStatus(request);
+  }
+
+  def getStringProperty(config: Map[String, AnyRef], key: String, defaultValue: String) : String = {
+    config.getOrElse(key, defaultValue).asInstanceOf[String]
+  }
+  def druidAlgorithm(reportConfig: ReportConfig)(implicit spark: SparkSession, sqlContext: SQLContext, fc: FrameworkContext, sc:SparkContext, config: JobConfig): RDD[DruidOutput]  ={
+    val queryDims = reportConfig.metrics.map { f =>
+      f.druidQuery.dimensions.getOrElse(List()).map(f => f.aliasName.getOrElse(f.fieldName))
+    }.distinct
+    if (queryDims.length > 1) throw new DruidConfigException("Query dimensions are not matching")
+
+    val interval = reportConfig.dateRange
+    val granularity = interval.granularity
+    val reportInterval = if (interval.staticInterval.nonEmpty) {
+      interval.staticInterval.get
+    } else if (interval.interval.nonEmpty) {
+      interval.interval.get
+    } else {
+      throw new DruidConfigException("Both staticInterval and interval cannot be missing. Either of them should be specified")
+    }
+    val metrics = reportConfig.metrics.map { f =>
+
+      val queryInterval = if (interval.staticInterval.isEmpty && interval.interval.nonEmpty) {
+        val dateRange = interval.interval.get
+        getDateRange(dateRange, interval.intervalSlider,f.druidQuery.dataSource)
+      } else
+        reportInterval
+
+      val queryConfig = if (granularity.nonEmpty)
+        JSONUtils.deserialize[Map[String, AnyRef]](JSONUtils.serialize(f.druidQuery)) ++ Map("intervalSlider" -> interval.intervalSlider,
+          "intervals" -> queryInterval, "granularity" -> granularity.get)
+      else
+        JSONUtils.deserialize[Map[String, AnyRef]](JSONUtils.serialize(f.druidQuery)) ++ Map("intervalSlider" -> interval.intervalSlider,
+          "intervals" -> queryInterval)
+      val data = DruidDataFetcher.getDruidData(JSONUtils.deserialize[DruidQueryModel](JSONUtils.serialize(queryConfig)))
+      data.map { x =>
+        val dataMap = JSONUtils.deserialize[Map[String, AnyRef]](x)
+        val key = dataMap.filter(m => (queryDims.flatten).contains(m._1)).values.map(f => f.toString).toList.sorted(Ordering.String.reverse).mkString(",")
+        (key, dataMap)
+
+      }
+    }
+    val finalResult = metrics.fold(sc.emptyRDD)(_ union _)
+    finalResult.map { f =>
+      DruidOutput(f._2)
+    }
+  }
+
+  def druidPostProcess(data: RDD[DruidOutput],request_id: String,reportConfig: ReportConfig, storageConfig:StorageConfig)(implicit spark: SparkSession, sqlContext: SQLContext, fc: FrameworkContext, sc:SparkContext, config: JobConfig): OnDemandDruidResponse = {
```

Comment:
Same as above. only have spark session as implicit variable

Reply:
implicit sc:SparkContext is required => as we use it ,to get the dataCount (val dataCount = sc.longAccumulator("DruidReportCount")) & getReportDF() requires it
       implicit sqlContext: SQLContext is required => as getReportDF() requires